packages/cadl-ranch-specs/http/azure/client-generator-core/format/main.tsp

```diff
@@ -0,0 +1,126 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+/**
+ * Describe scenarios for various combinations of unions.
+ */
+@supportedBy("dpg")
+@scenarioService("/azure/client-generator-core/format")
+namespace _Specs_.Azure.ClientGenerator.Core.Format;
+
+model Model {
+  @format("guid")
+  guidProperty?: string;
+
+  @format("url")
+  urlProperty?: string;
+
+  @format("uuid")
+  uuidProperty?: string;
+
+  @format("armId")
+  armIdProperty?: string;
+
+  @format("ipAddress")
+  ipAddressProperty?: string;
+
+  @format("azureLocation")
+  azureLocation?: string;
+
+  @format("etag")
+  etagProperty?: string;
+}
+
+model InputModel is Model;
+
+model OutputModel is Model;
+
+model RoundTripModel is Model;
+
+/**
+ * Consumes an InputModel with properties of various formats
+ */
+@route("/input")
+@post
+@scenario
+@scenarioDoc("""
+  Verify properties with various of formats can be processed in a request.
+
+  Expected request body:
+  ```json
+  {
+    "guidProperty": "3f2504e0-4f89-11d3-9a0c-0305e82c3301",
+    "urlProperty": "https://www.example.com/path/to/resource",
+    "uuidProperty": "3f2504e0-4f89-11d3-9a0c-0305e82c3301",
+    "armIdProperty": "/subscriptions/3f2504e0-4f89-11d3-9a0c-0305e82c3301/resourceGroups/test/providers/Microsoft.Compute/virtualMachines/name",
+    "ipAddressProperty": "192.168.1.1",
+    "azureLocation": "eastus",
+    "etagProperty": "33a64df551425fcc55e4d42a148795d9f25f89d4"
+  }
+  ```
+  """)
+op inputWithFormat(@body body: InputModel): void;
```

Comment:
nit, I don't think you need to repeat "WithFormat" in a test where everything is about format.

Reply:
sure. I will just change those to `input`, `output` and `roundtrip`